src/wazuh_db/wdb_agent.c

```diff
@@ -11,268 +11,180 @@
+    [SQL_UPDATE_AGENT_KEEPALIVE] = "global sql UPDATE agent SET last_keepalive = %lu WHERE id = %d;",
+    [SQL_DELETE_AGENT] = "global sql DELETE FROM agent WHERE id = %d;",
+    [SQL_SELECT_AGENT] = "global sql SELECT name FROM agent WHERE id = %d;",
+    [SQL_SELECT_AGENT_GROUP] = "global sql SELECT `group` FROM agent WHERE id = %d;",
+    [SQL_SELECT_AGENTS] = "global sql SELECT id FROM agent WHERE id != 0;",
+    [SQL_FIND_AGENT] = "global sql SELECT id FROM agent WHERE name = '%s' AND (register_ip = '%s' OR register_ip LIKE '%s' || '/_%');",
+    [SQL_SELECT_FIM_OFFSET] = "global sql SELECT fim_offset FROM agent WHERE id = %d;",
+    [SQL_SELECT_REG_OFFSET] = "global sql SELECT reg_offset FROM agent WHERE id = %d;",
+    [SQL_UPDATE_FIM_OFFSET] = "global sql UPDATE agent SET fim_offset = %lu WHERE id = %d;",
+    [SQL_UPDATE_REG_OFFSET] = "globL sql UPDATE agent SET reg_offset = %lu WHERE id = %d;",
+    [SQL_SELECT_AGENT_STATUS] = "global sql SELECT status FROM agent WHERE id = %d;",
+    [SQL_UPDATE_AGENT_STATUS] = "global sql UPDATE agent SET status = '%s' WHERE id = %d;",
+    [SQL_UPDATE_AGENT_GROUP] = "global sql UPDATE agent SET `group` = '%s' WHERE id = %d;",
+    [SQL_FIND_GROUP] = "global sql SELECT id FROM `group` WHERE name = '%s';",
+    [SQL_INSERT_AGENT_GROUP] = "global sql INSERT INTO `group` (name) VALUES('%s');",
+    [SQL_INSERT_AGENT_BELONG] = "global sql INSERT INTO belongs (id_group, id_agent) VALUES(%d, %d);",
+    [SQL_DELETE_AGENT_BELONG] = "global sql DELETE FROM belongs WHERE id_agent = %d",
+    [SQL_DELETE_GROUP_BELONG] = "global sql DELETE FROM belongs WHERE id_group = (SELECT id FROM 'group' WHERE name = '%s' );", 
+    [SQL_DELETE_GROUP] = "global sql DELETE FROM `group` WHERE name = '%s';",
+    [SQL_SELECT_GROUPS] = "global sql SELECT name FROM `group`;"
+ };
 
 /* Insert agent. It opens and closes the DB. Returns 0 on success or -1 on error. */
 int wdb_insert_agent(int id, const char *name, const char *ip, const char *register_ip, const char *key, const char *group, int keep_date) {
     int result = 0;
-    sqlite3_stmt *stmt;
-    const char * sql = SQL_INSERT_AGENT;
-    time_t date;
-
-    if (wdb_open_global() < 0)
-        return -1;
+    time_t date = 0;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
 
     if(keep_date) {
         date = get_agent_date_added(id);
     } else {
         time(&date);
     }
 
-    if (wdb_prepare(wdb_global, sql, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_int(stmt, 1, id);
-    sqlite3_bind_text(stmt, 2, name, -1, NULL);
-
-    if (ip){
-        sqlite3_bind_text(stmt, 3, ip, -1, NULL);
-    }
-    else {
-        sqlite3_bind_null(stmt, 3);
+    // If the strings are empty, insert NULL
+    if (!ip){
```

Comment:
If the condition `!ip` evaluates to `true`, it means that `ip == NULL`. So, the assignment within the `if` statement is not necessary at all.

Reply:
True.